### PR TITLE
ISPN-6438 Configure client SSL with properties

### PR DESCRIPTION
**Pull request by galderz**

### CODE REVIEW

**Comment by tristantarrant** [comment]
Looks good. This also needs adding to the docs and possibly an "secure jcache" example. Once https://github.com/infinispan/infinispan/pull/4219 is done we can provide one.

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/configuration/ConfigurationTest.java [hunk 1]
```diff
@@ -155,30 +270,52 @@ private void assertServer(String host, int port, ServerConfiguration serverCfg)
 
    private void validateConfiguration(Configuration configuration) {
       assertEquals(2, configuration.servers().size());
-      assertEquals(SomeAsyncExecutorFactory.class, configuration.asyncExecutorFactory().factoryClass());
-      assertEquals(SomeRequestBalancingStrategy.class, configuration.balancingStrategyClass());
-      assertEquals(SomeTransportfactory.class, configuration.transportFactory());
+      assertEqualsConfig(SomeAsyncExecutorFactory.class, ASYNC_EXECUTOR_FACTORY, configuration);
```
Comment (danberindei) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/configuration/ConfigurationTest.java]: Considering the OPTIONS map is only used here, wouldn't it have been simpler to "inline" the lambdas?

**Comment by tristantarrant** [comment]
We should probably also support "injecting" a pre-built SSLContext

**Comment by tristantarrant** [comment]
A few more properties, to enable authentication as well as encryption please:
authenticationEnabled (a boolean)
saslMechanism (a string, should support multiple values which could be comma-separated)
callbackHandler (an instance of javax.security.auth.callback.CallbackHandler)
serverName (a string)
clientSubject (an instance of javax.security.auth.Subject)
saslProperties (a map<String, String>, need to think how to represent this as a string, an instance would be fine)

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/configuration/ConfigurationTest.java (hunk 1 above)
Comment (galderz) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/configuration/ConfigurationTest.java]: The idea behind it is to be able to reuse them potentially in other client config tests.

**Comment by galderz** [comment]
@tristantarrant Authentication properties have been added.

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/Configuration.java [hunk 2]
```diff
@@ -202,4 +206,97 @@ public String toString() {
...
+      properties.setProperty(ConfigurationProperties.SERVER_LIST, servers.toString());
+
+      properties.setProperty(ConfigurationProperties.USE_SSL, Boolean.toString(security.ssl().enabled()));
+
+      if (security.ssl().keyStoreFileName() != null)
+         properties.setProperty(ConfigurationProperties.KEY_STORE_FILE_NAME, security.ssl().keyStoreFileName());
+
+      if (security.ssl().keyStorePassword() != null)
+         properties.setProperty(ConfigurationProperties.KEY_STORE_PASSWORD, new String(security.ssl().keyStorePassword()));
+
```
Comment (tristantarrant) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/Configuration.java]: I just pushed ISPN-6497 which adds one extra property: keyStoreCertificatePassword
Can you please add it @galderz. I promise it's the last one ;)

Comment (tristantarrant) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/Configuration.java]: I need to rectify my statement. Can you also make the callbackHandler with both with a String (and therefore initialize it with reflection) or an instance ?

**Comment by galderz** [comment]
@tristantarrant +1 to both comments, they're both included now.

**Comment by tristantarrant** [comment]
Pushed, thanks

**Pull request closed without merging**